src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -716,12 +602,19 @@ class module_elf : public module_impl
     : module_impl{ elf.get_cfg_uuid() }
     , m_elf(std::move(elf))
     , m_os_abi{ xrt_core::elf_int::get_elfio(m_elf).get_os_abi() }
-    , m_ctrlcodes{ initialize_column_ctrlcode(xrt_core::elf_int::get_elfio(m_elf)) }
-    , m_arg2patcher{ initialize_arg_patchers(xrt_core::elf_int::get_elfio(m_elf), m_ctrlcodes) }
-    , m_instr_buf{ initialize_instr_buf(xrt_core::elf_int::get_elfio(m_elf)) }
-    , m_ctrl_packet{ initialize_ctrl_packet(xrt_core::elf_int::get_elfio(m_elf)) }
-    , m_arg2patchers{ initialize_arg_patchers(xrt_core::elf_int::get_elfio(m_elf), m_instr_buf, m_ctrl_packet) }
-  {}
+  {
+    if ( m_os_abi == Elf_Amd_Aie2ps)
+    {
+      m_ctrlcodes = initialize_column_ctrlcode(xrt_core::elf_int::get_elfio(m_elf));
+      m_arg2patcher = initialize_arg_patchers(xrt_core::elf_int::get_elfio(m_elf), m_ctrlcodes);
+    }
+    else if ( m_os_abi == Elf_Amd_Aie2p)
+    {
```

Comment:
} else if (m_os_abi == Elf_Amd_Aie2p) {

Reply:
It's important  to me (actually not really) that `else` lines up with `if`.

```
if (cond1) {
  stmt1;
  stmt2;
}
else if (cond 2) {
  stmt3;
  stmt4
}  
```

Reply:
So user space code we follow this rule but in kernel space (Linuxe), we put "}" and "else/ elseif" in same line?

Reply:
I am sure there are many conventions that are different between userspace and kernel space :-) IMO is it much easier to scan code when things are lined up to the left, so this is the convention I follow in userspace.

Reply:
No problem. I just did not know this before. I will follow this rule when writing user space code.  Thanks.